### DISCUSSION TITLE
Method that replace onHosts() from dep6 to dep7

### DISCUSSION THREAD

**Discussion by splyy** [deployphp/deployer#2885] [Help needed]
Hello,

With v7.0.0-beta.26, How do you replace `onHosts()` method? 
I understand that you can use `select()`, but if that's the case I don't see how to translate it ?

```php
task('my:task, function() {
    run('...");
})->onHosts('demo');
```

Thanks,

**Comment by antonmedv** [comment]
Use select() https://deployer.org/docs/7.x/tasks#task-config

**Reply by splyy** [reply]
This is the solution I tried to use, but none of my ways of writing the parameter worked 🤔 

```php
->select('host=demo')
->select('hosts=demo')
->select('onHosts=demo')
```

That's why I've come to ask for help :D

**Reply by antonmedv** [reply]
I’m going to write a doc about it soon.

**Reply by leevigraham** [reply]
@antonmedv Did this get added to the docs? I can't seem to find it.

**Comment by mvuidev** [comment]
Hi,

I'm also trying to select a host like this : 

```php
localhost('local');
host('remote');

task('example', function(){
...
})->select('local');
```

What's the right way to make it work ?
Or maybe I misunderstood the usage of select().
